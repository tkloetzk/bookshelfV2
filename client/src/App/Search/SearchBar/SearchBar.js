import Grid from "@material-ui/core/Grid";
import React from "react";
import Textfield from "@material-ui/core/TextField";
import { makeStyles } from "@material-ui/styles";
import Button from "@material-ui/core/Button";
import isEmpty from "lodash/isEmpty";
import isIsbn from "is-isbn";
import forEach from "lodash/forEach";

const useStyles = makeStyles(theme => ({
  container: {
    padding: theme.spacing(2),
    margin: 0,
    width: "100%"
  }
}));

export default function SearchBar() {
  const classes = useStyles();
  const [searchedISBNs, setSearchedISBNs] = React.useState("");

  function handleSearch() {
    const isbns = searchedISBNs.split(/[\n, ]/).filter(v => v !== "");

    const promiseISBNs = [];
    forEach(isbns, isbn => {
      const formattedIsbn = isbn.replace(/[- ]/g, "");
      if (isIsbn.validate(formattedIsbn)) {
        promiseISBNs.push(formattedIsbn);
      }
    });

    // if (promiseISBNs.length) {
    //   Promise.all(map(promiseISBNs, (isbn) => {
    //     // getBook(isbn)
    //   }));
    // }
  }
  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      spacing={4}
      className={classes.container}
      data-testid="searchView"
    >
      <Grid item xs={8}>
        <Textfield
          value={searchedISBNs}
          onChange={event => setSearchedISBNs(event.target.value)}
          multiline
          variant="outlined"
          fullWidth
          inputProps={{ "data-testid": "searchBar" }}
        />
      </Grid>
      <Grid item xs={2}>
        <Button
          variant="outlined"
          color="primary"
          disabled={isEmpty(searchedISBNs)}
          data-testid="searchButton"
          onClick={handleSearch}
        >
          Search
        </Button>
      </Grid>
    </Grid>
  );
}
