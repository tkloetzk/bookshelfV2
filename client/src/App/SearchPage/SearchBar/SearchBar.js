import React from 'react'
import Textfield from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import isEmpty from 'lodash/isEmpty'
import isIsbn from 'is-isbn'
import forEach from 'lodash/forEach'
import search from '../../../services/searchService'
import union from 'lodash/union'
import compareDifferences from '../../../util/compareDifferences'
import Grid from '@material-ui/core/Grid'
import { useSelector } from 'react-redux'

export default function SearchPage({ setBooklist }) {
  const bookshelf = useSelector(state => state.bookshelf.bookshelf)
  const [searchedISBNs, setSearchedISBNs] = React.useState('')

  async function handleSearch() {
    const isbns = searchedISBNs.split(/[\n, ]/).filter(v => v !== '')

    const promiseISBNs = []
    forEach(isbns, isbn => {
      const formattedIsbn = isbn.replace(/[- ]/g, '')
      if (isIsbn.validate(formattedIsbn)) {
        promiseISBNs.push(formattedIsbn)
      }
    })

    const books = await search(union(promiseISBNs))

    //Could move this all into the util?
    var booklist = forEach(books, searchedBook => {
      return bookshelf.some(existingBook => {
        let searchedBookCopy = searchedBook
        if (searchedBook.isbn === existingBook.isbn) {
          searchedBookCopy.differences = compareDifferences(
            existingBook,
            searchedBookCopy,
            []
          )
        }
        return searchedBookCopy
      })
    })
    setBooklist(booklist)
  }

  return (
    <>
      <Grid item xs={8}>
        <Textfield
          value={searchedISBNs}
          onChange={event => setSearchedISBNs(event.target.value)}
          multiline
          variant="outlined"
          fullWidth
          inputProps={{ 'data-testid': 'searchBar' }}
        />
      </Grid>
      <Grid item xs={2}>
        <Button
          variant="outlined"
          color="primary"
          disabled={isEmpty(searchedISBNs)}
          data-testid="searchButton"
          onClick={handleSearch}
        >
          Search
        </Button>
      </Grid>
    </>
  )
}
