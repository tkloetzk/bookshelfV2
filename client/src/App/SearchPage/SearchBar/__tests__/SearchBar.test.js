import { MuiThemeProvider } from '@material-ui/core/styles'
import React from 'react'
import { fireEvent, render, wait } from '@testing-library/react'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import muiTheme from '../../../../config/themeConfig'
import SearchBar from '../SearchBar'
import '@testing-library/jest-dom/extend-expect'
import {
  addBookshelfService,
  updateBooksBookshelfService,
} from '../../../../services/bookshelfService'

const mockStore = configureMockStore()

jest.mock('../../../../services/bookshelfService')

describe('SearchBar', () => {
  let store
  let props
  let book
  let differences

  beforeEach(() => {
    store = mockStore({
      bookshelf: { bookshelf: [] },
    })
    store.dispatch = jest.fn()

    props = { setBooklist: jest.fn(), booklist: [] }
    book = {
      amazonAverageRating: 4.5,
      amazonRatingsCount: 553,
      categories: ['Family & Relationships'],
      description:
        'Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.',
      goodreadsAverageRating: 4,
      goodreadsRatingsCount: 40,
      isbn: '9781402218279',
      price: '',
      subtitle:
        'From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child',
      thumbnail:
        'http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api',
      title: 'The New Baby Answer Book',
      id: '1',
    }
    differences = [
      {
        currentValue: 3.9,
        key: 'amazonAverageRating',
        newValue: 4.5,
      },
    ]
    jest.clearAllMocks()
  })

  describe('render', () => {
    it('should render as expected', () => {
      const { asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar />
          </MuiThemeProvider>
        </Provider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
  describe('search button', () => {
    it('is enabled when the search bar has a value', async () => {
      const { getByTestId } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar />
          </MuiThemeProvider>
        </Provider>
      )
      await wait(() => {
        fireEvent.change(getByTestId('searchBar'), {
          target: { value: '123' },
        })
      })

      expect(getByTestId('searchButton')).not.toBeDisabled()
    })
    it('is disabled when the search bar is empty', async () => {
      const { getByTestId } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar />
          </MuiThemeProvider>
        </Provider>
      )
      await wait(() => {
        fireEvent.change(getByTestId('searchBar'), { target: { value: '' } })
      })

      expect(getByTestId('searchButton')).toBeDisabled()
    })
  })
  describe('handleSearch', () => {
    describe('validation', () => {
      it('does not call (search) if there are no promise isbns', async () => {
        const { getByTestId } = render(
          <Provider store={store}>
            <MuiThemeProvider theme={muiTheme}>
              <SearchBar {...props} />
            </MuiThemeProvider>
          </Provider>
        )
        await wait(() => {
          fireEvent.change(getByTestId('searchBar'), {
            target: { value: '123' },
          })
        })
        await wait(() => {
          fireEvent.click(getByTestId('searchButton'))
        })
        expect(props.setBooklist).not.toHaveBeenCalled()
      })
      it('does not include an isbn that is not valid', async () => {
        const { getByTestId } = render(
          <Provider store={store}>
            <MuiThemeProvider theme={muiTheme}>
              <SearchBar {...props} />
            </MuiThemeProvider>
          </Provider>
        )
        await wait(() => {
          fireEvent.change(getByTestId('searchBar'), {
            target: { value: '123,9781402218279' },
          })
        })
        await wait(() => {
          fireEvent.click(getByTestId('searchButton'))
        })
        expect(props.setBooklist.mock.calls).toMatchSnapshot()
      })
      it('can add a new book to the booklist on additional search', async () => {
        props = {
          ...props,
          booklist: [{ title: 'book1', isbn: '0123456789012' }],
        }

        const { getByTestId } = render(
          <Provider store={store}>
            <MuiThemeProvider theme={muiTheme}>
              <SearchBar {...props} />
            </MuiThemeProvider>
          </Provider>
        )
        await wait(() => {
          fireEvent.change(getByTestId('searchBar'), {
            target: { value: '9781402218279' },
          })
        })
        await wait(() => {
          fireEvent.click(getByTestId('searchButton'))
        })
        expect(props.setBooklist.mock.calls).toMatchSnapshot()
      })
      it('does not add a new book to the booklist if it already exists', async () => {
        props = { ...props, booklist: [{ isbn: '9781402218279' }] }

        const { getByTestId } = render(
          <Provider store={store}>
            <MuiThemeProvider theme={muiTheme}>
              <SearchBar {...props} />
            </MuiThemeProvider>
          </Provider>
        )
        await wait(() => {
          fireEvent.change(getByTestId('searchBar'), {
            target: { value: '9781402218279' },
          })
        })
        await wait(() => {
          fireEvent.click(getByTestId('searchButton'))
        })
        expect(props.setBooklist).not.toHaveBeenCalled()
      })
    })
    describe('setBooklist', () => {
      it('returns correct booklist if there are no books in bookshelf', async () => {
        const { getByTestId } = render(
          <Provider store={store}>
            <MuiThemeProvider theme={muiTheme}>
              <SearchBar {...props} />
            </MuiThemeProvider>
          </Provider>
        )
        await wait(() => {
          fireEvent.change(getByTestId('searchBar'), {
            target: { value: '9781402218279' },
          })
        })
        await wait(() => {
          fireEvent.click(getByTestId('searchButton'))
        })
        expect(props.setBooklist.mock.calls).toMatchSnapshot()
      })
      it('returns correct booklist if there are books, but no duplicates in bookshelf', async () => {
        store = mockStore({
          bookshelf: { bookshelf: [{ isbn: '1234' }] },
        })
        store.dispatch = jest.fn()

        const { getByTestId } = render(
          <Provider store={store}>
            <MuiThemeProvider theme={muiTheme}>
              <SearchBar {...props} />
            </MuiThemeProvider>
          </Provider>
        )
        await wait(() => {
          fireEvent.change(getByTestId('searchBar'), {
            target: { value: '9781402218279' },
          })
        })
        await wait(() => {
          fireEvent.click(getByTestId('searchButton'))
        })
        expect(props.setBooklist.mock.calls).toMatchSnapshot()
      })
      it('returns correct booklist if there are books, and a duplicate book in bookshelf with differences', async () => {
        store = mockStore({
          bookshelf: {
            bookshelf: [{ isbn: '9781402218279', amazonAverageRating: 3.9 }],
          },
        })
        store.dispatch = jest.fn()

        const { getByTestId } = render(
          <Provider store={store}>
            <MuiThemeProvider theme={muiTheme}>
              <SearchBar {...props} />
            </MuiThemeProvider>
          </Provider>
        )
        await wait(() => {
          fireEvent.change(getByTestId('searchBar'), {
            target: { value: '9781402218279' },
          })
        })
        await wait(() => {
          fireEvent.click(getByTestId('searchButton'))
        })
        expect(props.setBooklist.mock.calls).toMatchSnapshot()
      })
    })
  })
  describe('handleSave', () => {
    it('button is hidden if booklist is empty', () => {
      const { asFragment } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar {...props} />
          </MuiThemeProvider>
        </Provider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
    it('button is visible if booklist is not empty', () => {
      props = Object.assign({}, props, { booklist: [{ title: 'book title' }] })
      const { getByTestId } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar {...props} />
          </MuiThemeProvider>
        </Provider>
      )

      expect(getByTestId('saveButton')).toBeVisible()
    })
    it('calls addBookshelfService and does not call updateBookshelfService if there are no books with differences', async () => {
      props = Object.assign({}, props, { booklist: [book] })
      const { getByTestId } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar {...props} />
          </MuiThemeProvider>
        </Provider>
      )

      await wait(() => {
        fireEvent.click(getByTestId('saveButton'))
      })

      addBookshelfService.mockReturnValue(true)
      expect(updateBooksBookshelfService).not.toHaveBeenCalled()
      expect(addBookshelfService.mock.calls).toMatchSnapshot()
      expect(addBookshelfService).toHaveBeenCalledTimes(1)
      expect(store.dispatch).toHaveBeenCalledTimes(1)
    })
    it('does not call addBookshelfService but calls updateBooksBookshelfService with booklist containing only books with differences', async () => {
      book = Object.assign({}, book, { differences })
      props = Object.assign({}, props, { booklist: [book] })

      const { getByTestId } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar {...props} />
          </MuiThemeProvider>
        </Provider>
      )

      await wait(() => {
        fireEvent.click(getByTestId('saveButton'))
      })

      expect(updateBooksBookshelfService.mock.calls).toMatchSnapshot()
      expect(updateBooksBookshelfService).toHaveBeenCalledTimes(1)
      expect(addBookshelfService).not.toHaveBeenCalled()
      expect(store.dispatch).toHaveBeenCalledTimes(1)
    })
    it('calls both addBookshelfService and updateBooksBookshelfService with booklist containing books with and without differences', async () => {
      book = Object.assign({}, book, { differences })
      props = Object.assign({}, props, {
        booklist: [
          book,
          {
            isbn: '000',
            id: '999',
            differences: [
              {
                currentValue: 400,
                key: 'goodreadsRatingsCount',
                newValue: 500,
              },
              {
                currentValue: 100,
                key: 'amazonRatingsCount',
                newValue: 200,
              },
            ],
          },
          { title: 'book title', isbn: '1234' },
        ],
      })

      const { getByTestId } = render(
        <Provider store={store}>
          <MuiThemeProvider theme={muiTheme}>
            <SearchBar {...props} />
          </MuiThemeProvider>
        </Provider>
      )

      await wait(() => {
        fireEvent.click(getByTestId('saveButton'))
      })

      expect(updateBooksBookshelfService.mock.calls).toMatchSnapshot()
      expect(updateBooksBookshelfService).toHaveBeenCalledTimes(1)
      expect(addBookshelfService.mock.calls).toMatchSnapshot()
      expect(addBookshelfService).toHaveBeenCalledTimes(1)
      expect(store.dispatch).toHaveBeenCalledTimes(1)
    })
  })
})
