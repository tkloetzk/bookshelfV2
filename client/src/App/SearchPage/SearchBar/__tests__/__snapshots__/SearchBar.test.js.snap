// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar handleSave button is hidden if booklist is empty 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
      >
        <textarea
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          data-testid="searchBar"
          rows="1"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1972 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-1973"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-2"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
      data-testid="searchButton"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Search
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`SearchBar handleSave calls addBookshelfService and does not call updateBookshelfService if there are no books with differences 1`] = `
Array [
  Array [
    Array [
      Object {
        "amazonAverageRating": 4.5,
        "amazonRatingsCount": 553,
        "categories": Array [
          "Family & Relationships",
        ],
        "description": "Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.",
        "goodreadsAverageRating": 4,
        "goodreadsRatingsCount": 40,
        "id": "1",
        "isbn": "9781402218279",
        "price": "",
        "subtitle": "From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child",
        "thumbnail": "http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        "title": "The New Baby Answer Book",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSave calls both addBookshelfService and updateBooksBookshelfService with booklist containing books with and without differences 1`] = `
Array [
  Array [
    Array [
      Object {
        "fields": Object {
          "amazonAverageRating": 4.5,
        },
        "id": "1",
      },
      Object {
        "fields": Object {
          "amazonRatingsCount": 200,
          "goodreadsRatingsCount": 500,
        },
        "id": "999",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSave calls both addBookshelfService and updateBooksBookshelfService with booklist containing books with and without differences 2`] = `
Array [
  Array [
    Array [
      Object {
        "isbn": "1234",
        "title": "book title",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSave does not call addBookshelfService but calls updateBooksBookshelfService with booklist containing only books with differences 1`] = `
Array [
  Array [
    Array [
      Object {
        "fields": Object {
          "amazonAverageRating": 4.5,
        },
        "id": "1",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSearch setBooklist returns correct booklist if there are books, and a duplicate book in bookshelf with differences 1`] = `
Array [
  Array [
    Array [
      Object {
        "amazonAverageRating": 4.5,
        "amazonRatingsCount": 553,
        "categories": Array [
          "Family & Relationships",
        ],
        "description": "Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.",
        "differences": Array [
          Object {
            "currentValue": 3.9,
            "key": "amazonAverageRating",
            "newValue": 4.5,
          },
        ],
        "goodreadsAverageRating": 4,
        "goodreadsRatingsCount": 40,
        "id": undefined,
        "isbn": "9781402218279",
        "price": "",
        "subtitle": "From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child",
        "thumbnail": "http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        "title": "The New Baby Answer Book",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSearch setBooklist returns correct booklist if there are books, but no duplicates in bookshelf 1`] = `
Array [
  Array [
    Array [
      Object {
        "amazonAverageRating": 4.5,
        "amazonRatingsCount": 553,
        "categories": Array [
          "Family & Relationships",
        ],
        "description": "Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.",
        "goodreadsAverageRating": 4,
        "goodreadsRatingsCount": 40,
        "isbn": "9781402218279",
        "price": "",
        "subtitle": "From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child",
        "thumbnail": "http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        "title": "The New Baby Answer Book",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSearch setBooklist returns correct booklist if there are no books in bookshelf 1`] = `
Array [
  Array [
    Array [
      Object {
        "amazonAverageRating": 4.5,
        "amazonRatingsCount": 553,
        "categories": Array [
          "Family & Relationships",
        ],
        "description": "Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.",
        "goodreadsAverageRating": 4,
        "goodreadsRatingsCount": 40,
        "isbn": "9781402218279",
        "price": "",
        "subtitle": "From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child",
        "thumbnail": "http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        "title": "The New Baby Answer Book",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSearch validation can add a new book to the booklist on additional search 1`] = `
Array [
  Array [
    Array [
      Object {
        "isbn": "0123456789012",
        "title": "book1",
      },
      Object {
        "amazonAverageRating": 4.5,
        "amazonRatingsCount": 553,
        "categories": Array [
          "Family & Relationships",
        ],
        "description": "Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.",
        "goodreadsAverageRating": 4,
        "goodreadsRatingsCount": 40,
        "isbn": "9781402218279",
        "price": "",
        "subtitle": "From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child",
        "thumbnail": "http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        "title": "The New Baby Answer Book",
      },
    ],
  ],
]
`;

exports[`SearchBar handleSearch validation does not include an isbn that is not valid 1`] = `
Array [
  Array [
    Array [
      Object {
        "amazonAverageRating": 4.5,
        "amazonRatingsCount": 553,
        "categories": Array [
          "Family & Relationships",
        ],
        "description": "Answers the most important parenting questions about raising children from birth through kindergarten, covering such topics as discipline, sleeping, day care, safety, independence, and feeding.",
        "goodreadsAverageRating": 4,
        "goodreadsRatingsCount": 40,
        "isbn": "9781402218279",
        "price": "",
        "subtitle": "From Birth to Kindergarten, Answers to the Top 150 Questions about Raising a Young Child",
        "thumbnail": "http://books.google.com/books/content?id=oD0omQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        "title": "The New Baby Answer Book",
      },
    ],
  ],
]
`;

exports[`SearchBar render should render as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
      >
        <textarea
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          data-testid="searchBar"
          rows="1"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; width: 100%;"
          tabindex="-1"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-144 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-145"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-2"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
      data-testid="searchButton"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Search
      </span>
    </button>
  </div>
</DocumentFragment>
`;
